### PR TITLE
py-eli5: Submission

### PR DESCRIPTION
**Pull request by essandess**
py-eli5: Submission

* py-eli5: Submission
* py-singledispatch: Add Python 3

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10.x
Xcode 8.x

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [x] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: python/py-eli5/Portfile [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+                    port:py${python.version}-jinja2 \
+                    port:py${python.version}-numpy \
+                    port:py${python.version}-scikit-learn \
+                    port:py${python.version}-scipy \
+                    port:py${python.version}-singledispatch \
+                    port:py${python.version}-tabulate
+
+    depends_test-append \
+                    port:py${python.version}-tensorflow \
+                    port:py${python.version}-tox
```
Comment (reneeotten) [inline python/py-eli5/Portfile]: you seem to be missing dependencies here comparing to upstream's list

File: python/py-eli5/Portfile [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        set docdir ${prefix}/share/doc/${subport}
+        xinstall -d \
+            ${destroot}${docdir}
+        xinstall -m 0644 -W ${worksrcpath} README.rst \
+            ${destroot}${docdir}
+    }
+
+    test.run        yes
+    test.cmd        tox-${python.branch}
+    test.env-append PYTHONPATH=${worksrcpath}/build/lib
```
Comment (reneeotten) [inline python/py-eli5/Portfile]: did you try the tests locally? You cannot just use upstream's ```tox.ini``` file here, that will try to run all different combinations they have selected. For example, if you run ```sudo port -dv test py37-eli5``` it will start with PY27 and, in fact, start downloading/downloading dependencies with ```pip```. 

That is not how things should happen. You can try to figure out how to run the tests directly with ```pytest``` (likely you'll need to add some test dependencies). Please only add a test phase if you've actually figured out how to run it and if you make them pass.

File: python/py-eli5/Portfile [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+categories-append   math
+license             MIT
+platforms           darwin
+maintainers         nomaintainer
+
+description         A library for debugging/inspecting machine learning\
+                    classifiers and explaining their predictions
+
+long_description    ${description}.\
```
[the remark this answers is not in the document]
Comment (essandess) [reply python/py-eli5/Portfile]: Thanks done.

File: python/py-eli5/Portfile [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+                    port:py${python.version}-setuptools
+
+    depends_lib-append \
+                    port:py${python.version}-attrs \
+                    port:py${python.version}-graphviz \
+                    port:py${python.version}-jinja2 \
+                    port:py${python.version}-numpy \
+                    port:py${python.version}-scikit-learn \
+                    port:py${python.version}-scipy \
+                    port:py${python.version}-singledispatch \
```
[the remark this answers is not in the document]
Comment (essandess) [reply python/py-eli5/Portfile]: Thanks done.

File: python/py-singledispatch/Portfile [hunk 5]
```diff
@@ -11,7 +11,7 @@ platforms           darwin
 license             MIT
 supported_archs     noarch
 
-python.versions     27
+python.versions     27 35 36 37 38
```
[the remark this answers is not in the document]
Comment (essandess) [reply python/py-singledispatch/Portfile]: Thanks done.

File: python/py-eli5/Portfile (hunk 2 above)
Comment (essandess) [reply python/py-eli5/Portfile]: Yes, the local tests issued a warning to use tox. They mostly run, with a few errors. I'll just delete the whole things.

File: python/py-eli5/Portfile (hunk 1 above)
Comment (essandess) [reply python/py-eli5/Portfile]: I just deleted the whole test block.
